### PR TITLE
feature: delete product

### PR DESCRIPTION
**Pull request by steven-aldo-sutanto**
## Description
This Pull Request is to create a new endpoint `DELETE api/v1/product/{product_id}` for deleting a product by passing the `product_id`.

## Unit Test
image

**Related Link:** https://mitrais-dev.atlassian.net/browse/MTK-160

### CODE REVIEW

**Review by fazarrahman — APPROVED** [review]

File: service/service.go [hunk 1]
```diff
@@ -485,6 +486,34 @@ func (s *service) UpsertProductType(ctx context.Context, prodType *prodEntity.Pr
...
+	if err := s.imageRepository.RemoveImage(ctx, removeProdImageIds, "product", userId.String()); err != nil {
+		s.productRepository.TransactionRollback()
+		return util.NewError(codes.Internal, errPb.StoreErrorCode_ERROR_WHEN_REMOVING_IMAGE_FROM_STORAGE.String(), "Error saat menghapus gambar dari penyimpanan : "+err.Error())
+	}
+	if err := s.productRepository.DeleteProductImages(ctx, prodImages); err != nil {
+		s.productRepository.TransactionRollback()
+		return util.NewError(codes.Internal, errPb.StoreErrorCode_ERROR_WHEN_DELETING_PRODUCT_IMAGE.String(), "Error saat menghapus gambar produk : "+err.Error())
+	}
+
+	err = s.productRepository.DeleteProductById(ctx, id)
```
Comment (fazarrahman) [inline service/service.go:512]: Please handle. if there is error, transaction will be rolledback

File: service/service.go [hunk 2]
```diff
@@ -485,6 +486,34 @@ func (s *service) UpsertProductType(ctx context.Context, prodType *prodEntity.Pr
...
+
```
Comment (fazarrahman) [inline service/service.go:513]: please call s.productRepository.Commit()

File: domain/product/repository/postgres/postgres.go [hunk 3]
```diff
@@ -158,6 +158,42 @@ func (p *Postgres) DeleteProductImages(ctx context.Context, productImages []*ent
 	return nil
 }
 
+func (p *Postgres) DeleteProductById(ctx context.Context, id uuid.UUID) error {
+	err := p.db.WithContext(ctx).Transaction(func(tx *gorm.DB) error {
+		err := tx.Model(&entity.Product{}).Where("id = ?", id.String()).Updates(map[string]interface{}{
```
Comment (fazarrahman) [inline domain/product/repository/postgres/postgres.go:163]: no need to update sale status to false

File: service/service.go (hunk 1 above)
Comment (steven-aldo-sutanto) [reply service/service.go]: Has been updated to handle error

File: service/service.go (hunk 2 above)
Comment (steven-aldo-sutanto) [reply service/service.go:523]: s.productRepository.TransactionCommit() is called to make sure data is committed at the end of the function.

File: domain/product/repository/postgres/postgres.go (hunk 3 above)
Comment (steven-aldo-sutanto) [reply domain/product/repository/postgres/postgres.go]: Removed the update `sale_status`.

**Review by fazarrahman — APPROVED** [review]

**Pull request merged**